src/core/moves/mixture/OrderedEventTimeSlideProposal.cpp

```diff
@@ -0,0 +1,272 @@
+    OrderedEventTimes &event_variable   = event_var->getValue();
+	const std::set<double>& event_times = event_variable.getEventTimes();
+	size_t num_events = event_variable.size();
+
+	// if there are zero events, abort
+	if ( num_events == 0 )
+	{
+		failed = true;
+		return RbConstants::Double::neginf;
+	}
+
+    // choose an element to update
+	size_t index = size_t(rng->uniform01() * num_events);
+
+	// get the value of the chosen element
+	std::set<double>::const_iterator the_event = event_times.begin();
+	std::advance(the_event, index);
+	old_time = *the_event;
+
+	// draw a new value for the chosen element
+	double u = rng->uniform01();
+	double shift = ( delta * ( u - 0.5 ) );
+	new_time = old_time + shift;
+
+	// reflect back from negative values (hastings ratio is 1)
+	if ( new_time < 0.0 )
+	{
+		new_time *= -1.0;
+	}
+	// TODO: consider implementing an upper-bound reflection, as well
+
+	// update the value in the variable
+	failed = !event_variable.changeEventTime(old_time, new_time);
+	if ( failed == true )
+	{
+		return RbConstants::Double::neginf;
+	}
+
+	// make sure the dag node is updated
+	event_var->touch(true);
+
+	// update all the children
+	const std::vector<DagNode*>& children = event_var->getChildren();
+	for(std::vector<DagNode*>::const_iterator it = children.begin(); it != children.end(); ++it)
+	{
+		if ( (*it)->isStochastic() == true )
+		{
+			AbstractEventsDistribution* dist = dynamic_cast<AbstractEventsDistribution *>( &(*it)->getDistribution() );
+			if ( dist != NULL )
+			{ // this child is an event distribution
+				dist->changeEventTime(old_time, new_time);
+				(*it)->touch(true);
+			}
+		}
+	}
+
+	return 0.0;
+}
+
+
+/**
+ *
+ */
+void OrderedEventTimeSlideProposal::prepareProposal( void )
+{
+    
+}
+
+
+/**
+ * Print the summary of the Proposal.
+ *
+ * The summary just contains the current value of the tuning parameter.
+ * It is printed to the stream that it passed in.
+ *
+ * \param[in]     o     The stream to which we print the summary.
+ */
+void OrderedEventTimeSlideProposal::printParameterSummary(std::ostream &o, bool name_only) const
+{
+    o << "delta = ";
+    if (name_only == false)
+    {
+        o << delta;
+    }
+}
+
+
+/**
+ * Reject the Proposal.
+ *
+ * Since the Proposal stores the previous value and it is the only place
+ * where complex undo operations are known/implement, we need to revert
+ * the value of the variable/DAG-node to its original value.
+ */
+void OrderedEventTimeSlideProposal::undoProposal( void )
+{
+	if (failed == false)
+	{
+	    // get the variable
+	    OrderedEventTimes &event_variable = event_var->getValue();
+
+	    // reset the time
+	    event_variable.changeEventTime(new_time, old_time);
+
+		// make sure the dag node is updated
+		event_var->touch(true);
+
+		// reset all the children
+		const std::vector<DagNode*>& children = event_var->getChildren();
+		for(std::vector<DagNode*>::const_iterator it = children.begin(); it != children.end(); ++it)
```

Comment:
Perhaps `for(auto& child: event_var->getChildren())` and replace `(*it)` with `child`